boot/bootutil/include/bootutil/sha256.h

```diff
@@ -31,13 +31,12 @@
 
 #include "mcuboot_config/mcuboot_config.h"
 
-#if (defined(MCUBOOT_USE_MBED_TLS) && defined(MCUBOOT_USE_TINYCRYPT))\
-    ||(defined MCUBOOT_USE_MBED_TLS) && defined(MCUBOOT_USE_CC310)\
-    ||(defined(MCUBOOT_USE_TINYCRYPT) && defined(MCUBOOT_USE_CC310))
-    #error "Cannot define CC310, MBED_TLS and TINYCRYPT"
-#endif
-
-#if !defined(MCUBOOT_USE_MBED_TLS) && !defined(MCUBOOT_USE_TINYCRYPT) && !defined(MCUBOOT_USE_CC310)
+#define NUM_CRYPTO_BACKENDS (defined(MCUBOOT_USE_MBED_TLS) +\
+			     defined(MCUBOOT_USE_TINYCRYPT) +\
+			     defined(MCUBOOT_USE_CC310))
+#if NUM_CRYPTO_BACKENDS > 1
+    #error "Cannot define pairs of CC310, MBED_TLS, and TINYCRYPT"
+#elif NUM_CRYPTO_BACKENDS == 0
     #error "One of CC310, MBED_TLS or TINYCRYPT must be defined"
```

Comment:
This message would also apply to the first test since it's "One of ..."

Reply:
Fixed

Reply:
Maybe I should rephrase it to "Only one"